db/migrate/20170927135007_migrate_dialog_field_associations_to_use_new_relationship.rb

```diff
@@ -0,0 +1,46 @@
+class MigrateDialogFieldAssociationsToUseNewRelationship < ActiveRecord::Migration[5.0]
+  class DialogFieldAssociation < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogField < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class Dialog < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogTab < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogGroup < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+
+  def up
+    dialog_fields_with_associations = DialogField.all.select { |df| df.auto_refresh || df.trigger_auto_refresh }.sort { |n| -n.position }
+    triggers = dialog_fields_with_associations.select(&:trigger_auto_refresh)
+    responders = dialog_fields_with_associations.select(&:auto_refresh)
+    triggers.each_with_index do |t, index|
+      responder_fields_for_t = if triggers[index + 1]
+                                 responders.select { |r| r.position > t.position && r.position <= triggers[index + 1].position }.pluck(:id)
+                               else
+                                 responders.select { |r| r.position > t.position }.pluck(:id)
+                               end
+      trigger_field_dialog_id = DialogTab.find(DialogGroup.find(t.dialog_group_id).dialog_tab_id).dialog_id
+      responder_fields_for_t.each do |responder|
+        respond_field_dialog_id = DialogTab.find(DialogGroup.find(DialogField.find(responder).dialog_group_id).dialog_tab_id).dialog_id
+        if trigger_field_dialog_id == respond_field_dialog_id
+          DialogFieldAssociation.create(:trigger_id => t.id, :respond_id => responder)
+        end
```

Comment:
I don't think we have to change this, but at first glance it was kinda weird to see "if the trigger id equals the responder id, let's create an association", which seems wrong. Maybe a variable name change? Same with donig `:respond_id => responder` *looks* like you're setting it to the object even though you're iterating through a list of ids.

Another way to do it instead of taking *all* the dialog fields right from the start is to iterate through the dialogs themselves and then you can just do `dialog.dialog_fields` instead of having to make sure that the dialog ids match for the fields you're comparing?

I think if we want to do it this way, that's ok, but there needs to be a spec that tests the fact that the fields need to belong to the same dialog.

Reply:
After discussions with Fryguy and bdunne it seemed easier to go by field first because otherwise the tests have to include all the information about dialogs and tabs and groups and those relationships whereas all that's needed here is the class definition.